src/character.cpp

```diff
@@ -4046,7 +4046,9 @@ void Character::mod_stored_calories( int ncal, const bool ignore_weariness )
     if( !ignore_weariness ) {
         activity_history.calorie_adjust( ncal );
     }
-    set_stored_calories( stored_calories + ncal );
+    if( ncal < 0 || ncal + stored_calories > 0 ) {
```

Comment:
Integer overflow is undefined so even if `ncal + stored_caloreis > 0` it could still have overflowed (for example if the target system uses saturation instead of wrap around for overflow).